### ISSUE TITLE
AssertionError during type argument inference

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#404]
```
What steps will reproduce the problem?
Compile the attached Java class using `javac -processor 
org.checkerframework.checker.nullness.NullnessChecker -version -verbose 
-AprintErrorStack -AprintAllQualifiers StringUtil.java`

What is the expected output? What do you see instead?
I expect the compilation to complete successfully. Instead, an AssertionError 
is thrown. See the attached compiler output.

What version of the product are you using? On what operating system?
I built the "trunk" versions of jsr308-langtools, annotation-tools and 
checker-framework from source, as described in the manual, using jdk1.8.0_31. 
I'm running linux.

Please provide any additional information below.
Didn't try to debug the issue beyond providing this minimal (?) test case, 
sorry.
```

Original issue reported on code.google.com by `stephan...@gmail.com` on 28 Feb 2015 at 10:53

Attachments:
- [StringUtil.java](https://storage.googleapis.com/google-code-attachments/checker-framework/issue-404/comment-0/StringUtil.java)
- [StringUtil-compiler-output.txt](https://storage.googleapis.com/google-code-attachments/checker-framework/issue-404/comment-0/StringUtil-compiler-output.txt)

**Comment by smillst** [comment]
This false positives should go away once #979 is fixed. (These test cases will be added to that issue.)
